Add search student by id or name feature to gradebook system #51

CalebBujauskas opened the pull request:
**Features**

1. Users can search for students within a course

**Code**

1. Added search_student function to gradebook.py
2. Added search students selection as option number 5 in main.py
3. Added tests for search_student function

**Brief Code Overview**

1. _caleb_test_search_student.py_ contains all tests for search_student function within gradebook.py

**To test**

1.   Pull this branch
2.   Run **main.py** using _"python"_ or _"python3"_ command
3.   Login has a valid instructor
4.   Enter valid course
5.   Enter "5" for _search students_ option
6.   Enter search query (either student name/id or letter/letters that may be apart of student names
7.   Follow prompts
8.   Repeat searches until you enter _"back"_ when prompted
9.   Enter _"x"_ to logout
10. Enter _"enter"_ to continue
11. Enter _"q"_ to quit program
12. Change directory to **_GradebookSystem/tests_**
13. Run **caleb_test_search_student.py** using _"pytest"_

CalebBujauskas commented:
**_This is for issue #51_**

sjiang1 commented:
Overall, good job!

In the issue, it talks about:
> Use this search feature before adding or editing grades to save time.

I don't see the branch implements that. Please add this functionality to your search option.

Also, update your branch with the new commits in the main branch.

Due to the time limit, I will check the rest of code after your updates.

sjiang1 reviewed (changes requested):
Please see my previous comment.

sjiang1 commented:
Also, tests failed.

sjiang1 reviewed (approved):
All good! Congrats!
I removed your name from the test file name. Please see my last commit in your branch.
Really want to merge it.

CalebBujauskas commented:
**Code**

1.  Implemented the _search_student_ function properly by integrating the feature within both _add_grade_ and _edit_grade_ functions.
2.  Updated necessary test files to include test input for the _search_student_ integration.
3.  Created a _helper_search_student_ function to properly call the _search_student_ function.

**To Test**

1. Pull this branch
2. Run main.py using _"python"_ or _"python3"_ command
3. Login has a valid instructor
4. Enter valid course
5. Select option _1_ or _2_
6. Enter _"y"_ or _"n"_ to search for student
7. Repeat search as much as you'd like
8. Enter _"back"_ to exit search
9. Continue with prompts
10. Logout is same as previous test instructions
11. Switch to **Gradebook/tests/ directory** 
12. Run all tests

CalebBujauskas commented:
Thank you very much!
